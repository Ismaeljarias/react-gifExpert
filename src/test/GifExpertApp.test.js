import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("<GifExpertApp /> Tests", () => {
  test("should show <GifExpertApp /> correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a categories list", () => {
    const categories = ["One Punch", "My hero academy"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
