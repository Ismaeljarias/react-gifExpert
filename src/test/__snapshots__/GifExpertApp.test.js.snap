// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> Tests should show <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp /> Tests should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="My hero academy"
      key="My hero academy"
    />
  </ol>
</Fragment>
`;
